lib/index.js

```diff
@@ -196,7 +196,9 @@ Queue.prototype._processHead = function() {
   var now = this._schedule.now();
   var toRun = [];
 
-  while (queue.length && queue[0].time <= now) {
+  var inProgressSize = Object.keys(inProgress).length;
+
+  while (queue.length && queue[0].time <= now && inProgressSize++ < self.maxItems) {
```

Comment:
why the `++` in `inProgressSize++`?

Reply:
oh it's because the loop adds to the inProgress map - makes sense!

Reply:
Because we add an item to `inProgress` inside of the loop, so the length grows. It's a slightly faster version of : 
```js
while (queue.length && queue[0].time <= now && Object.keys(inProgress).length < self.maxItems)
```